crates/astria-sequencer/src/state_ext.rs

```diff
@@ -243,3 +243,298 @@ pub(crate) trait StateWriteExt: StateWrite {
 }
 
 impl<T: StateWrite> StateWriteExt for T {}
+
+#[cfg(test)]
+mod test {
+    use cnidarium::StateDelta;
+    use tendermint::Time;
+
+    use super::{
+        StateReadExt as _,
+        StateWriteExt as _,
+    };
+
+    #[tokio::test]
+    async fn chain_id() {
+        let storage = cnidarium::TempStorage::new()
+            .await
+            .expect("failed to create temp storage backing chain state");
+        let snapshot = storage.latest_snapshot();
+        let mut state = StateDelta::new(snapshot);
+
+        // doesn't exist at first
+        assert!(state.get_chain_id().await.is_err());
+
+        // can write new
+        let chain_id_orig = "test-chain-orig";
+        state.put_chain_id(chain_id_orig.to_string());
+        assert!(state.get_chain_id().await.unwrap() == chain_id_orig);
```

Comment:
```suggestion
        assert_eq!(state.get_chain_id().await.unwrap(), chain_id_orig);
```
This will give you better diagnostics in case smth went wrong.

Also, this might be a good use `expect("a chain ID was written and must exist in the database")`